R/make_phylo_tree.R

```diff
@@ -0,0 +1,185 @@
+#' Build a `phyloTree` object for geneplast
+#'
+#'
+#'
+#' @param sspids a vector or data frame containing NCBI Taxon IDs from the species of interest.
+#' @param newick a rooted phylogenetic tree in Newick format.
+#' @param verbose a logical value specifying whether or not to display detailed messages.
+#' @return An object of class "phylo".
+#' @export
+#' @author Danilo O Imparato
+make.phyloTree <- function(sspids=NULL, newick=NULL, verbose=TRUE){
+    check_args(sspids, newick)
+    if(!is.null(newick)) {
+        phylo <- treeio::read.newick(newick)
+        return(phylo)
+    }
```

Comment:
Realmente acho que não deveria existir esse argumento `newick` aqui. Esse comportamento faria bem mais sentido no próprio argumento `phyloTree` do `geneplast::preprocess`. Em `geneplast::preprocess`, `phyloTree` aceitaria tanto um objeto `phylo` ou um `character` com o caminho do newick.

```suggestion
make.phyloTree <- function(sspids=NULL, verbose=TRUE){
    check_args(sspids)
```

Reply:
Concordo, Danilo!
Originalmente estava implementado dessa maneira. Foi Rodrigo quem pediu que fosse incluído esse segundo argumento.
Vou falar com ele hoje ASAP pra gente resolver isso.